src/main/java/uk/gov/companieshouse/overseasentitiesapi/mapper/impl/TrustDataMapperImpl.java

```diff
@@ -0,0 +1,222 @@
+                Integer.parseInt(trustDataDto.getTrustCreationDateDay())
+            ));
+        trustDataDao.setUnableToObtainAllTrustInfo(Boolean.parseBoolean(trustDataDto.getUnableToObtainAllTrustInfo()));
+        setBeneficialOwners(trustDataDto, trustDataDao);
+
+        return trustDataDao;
+    }
+
+    private void setBeneficialOwners(TrustDataDto trustDataDto, TrustDataDao trustDataDao) throws ServiceException {
+        List<BeneficialOwnerDao> grantors = new ArrayList<>();
+        List<BeneficialOwnerDao> settlers = new ArrayList<>();
+        List<BeneficialOwnerDao> beneficiaries = new ArrayList<>();
+        List<BeneficialOwnerDao> interestedPersons = new ArrayList<>();
+
+        for (IndividualDto individualDto : trustDataDto.getIndividualDtos()) {
+            switch (individualDto.getType()) {
+                case INDIVIDUAL_GRANTORS:
+                    grantors.add(individualDtoToBeneficialOwnerDao(individualDto));
+                    break;
+                case INDIVIDUAL_SETTLERS:
+                    settlers.add(individualDtoToBeneficialOwnerDao(individualDto));
+                    break;
+                case INDIVIDUAL_BENEFICIARIES:
+                    beneficiaries.add(individualDtoToBeneficialOwnerDao(individualDto));
+                    break;
+                case INDIVIDUAL_INTERESTED_PERSONS:
+                    interestedPersons.add(individualDtoToBeneficialOwnerDao(individualDto));
+                    break;
+                default:
+                    throw new ServiceException("Invalid beneficial owner type received: " + individualDto.getType());
+            }
+        }
+
+        if (trustDataDto.getCorporateDtos() != null && !trustDataDto.getCorporateDtos().isEmpty()) {
+            for (CorporateDto corporateDto : trustDataDto.getCorporateDtos()) {
+                switch (corporateDto.getType()) {
+                    case CORPORATE_GRANTORS:
+                        grantors.add(corporateDtoToBeneficialOwner(corporateDto));
+                        break;
+                    case CORPORATE_SETTLERS:
+                        settlers.add(corporateDtoToBeneficialOwner(corporateDto));
+                        break;
+                    case CORPORATE_BENEFICIARIES:
+                        beneficiaries.add(corporateDtoToBeneficialOwner(corporateDto));
+                        break;
+                    case CORPORATE_INTERESTED_PERSONS:
+                        interestedPersons.add(corporateDtoToBeneficialOwner(corporateDto));
+                        break;
+                    default:
+                        throw new ServiceException("Invalid beneficial owner type received: " + corporateDto.getType());
+                }
+            }
+        }
+
+        List<BeneficialOwnerDao> historicalBoDtoList = new ArrayList<>();
+        if (trustDataDto.getHistoricalBoDtos() != null && !trustDataDto.getHistoricalBoDtos().isEmpty()) {
```

Comment:
Suggest using the `Objects.nonNull()` method here rather than '!= null' which just reads a little nicer.